src/containers/QuoteCreate/QuoteCreate.tsx

```diff
@@ -0,0 +1,136 @@
+import * as sdk from '@kiltprotocol/sdk-js'
+import React from 'react'
+import { connect, MapStateToProps } from 'react-redux'
+import { withRouter, RouteComponentProps } from 'react-router'
+import * as common from 'schema-based-json-editor'
+import { IMyIdentity } from '../../types/Contact'
+import { State as ReduxState } from '../../state/PersistentStore'
+import SchemaEditor from '../../components/SchemaEditor/SchemaEditor'
+import * as Quotes from '../../state/ducks/Quotes'
+import * as Wallet from '../../state/ducks/Wallet'
+
+import './QuoteCreate.scss'
+
+type StateProps = {
+  selectedIdentity: IMyIdentity
+}
+
+type DispatchProps = {
+  saveQuote: (
+    attesterSignedQuote: sdk.IQuoteAttesterSigned,
+    claimerIdentity: string
+  ) => void
+}
+
+type OwnProps = {
+  cTypeHash?: sdk.ICType['hash']
+  claimerAddress?: string
+  attesterAddress?: string
+  onCancel?: () => void
+  quoteId: (quoteId: Quotes.Entry['quoteId']) => void
+}
+
+type Props = RouteComponentProps<{ quoteId: Quotes.Entry['quoteId'] }> &
+  StateProps &
+  OwnProps &
+  DispatchProps
+
+type State = {
+  quote?: sdk.IQuote
+  initialValue?: object
+}
+
+class QuoteCreate extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    const { attesterAddress, cTypeHash } = this.props
+    this.state = {
+      initialValue: {
+        attesterAddress,
+        cTypeHash,
+      },
+    }
+    this.handleCancel = this.handleCancel.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.updateValue = this.updateValue.bind(this)
+  }
+
+  public updateValue = (value: sdk.IQuote): void => {
+    const quote = value
+    // Need to add an input for dates to actually have a selection of the dates.
+    quote.timeframe = new Date()
```

Comment:
Yes, I was meant to ask Timo about this. But the whole AntiCov app came about. 

But I would like to have a date picker for sure and it was a way to test. You are right, they do come out differently. one as a date and the other as a `string` value. I presume that one of the claims you sent and the other you didn't. 

If you didn't send the `RequestForTerms` to the claimer, it would produce the `string`.

Sending it adds the timeframe as `Date`.

Obviously this isn't the attained effect. I need to save the `Quote` only if they have sent the message for the `RequestForTerms`. Thank you! I will try and solve this.

Reply:
Again, we can pitch the idea of a date picker to Timo.

Reply:
Done.